code/datums/ai_laws/ai_laws.dm

```diff
@@ -0,0 +1,318 @@
+#define AI_LAWS_ASIMOV "asimov"
+
+/datum/ai_laws
+	var/name = "Unknown Laws"
+	var/zeroth = null
+	var/zeroth_borg = null
+	var/list/inherent = list()
+	var/list/supplied = list()
+	var/list/ion = list()
+	var/list/hacked = list()
+	var/mob/living/silicon/owner
+	var/id = DEFAULT_AI_LAWID
+
+/datum/ai_laws/Destroy(force=FALSE, ...)
+	if(!QDELETED(owner)) //Stopgap to help with laws randomly being lost. This stack_trace will hopefully help find the real issues.
+		if(force) //Unless we're forced...
+			stack_trace("AI law datum for [owner] has been forcefully destroyed incorrectly; the owner variable should be cleared first!")
+			return ..()
+		stack_trace("AI law datum for [owner] has ignored Destroy() call; the owner variable must be cleared first!")
+		return QDEL_HINT_LETMELIVE
+	owner = null
+	return ..()
+
+/datum/ai_laws/proc/lawid_to_type(lawid)
+	var/all_ai_laws = subtypesof(/datum/ai_laws)
+	for(var/al in all_ai_laws)
+		var/datum/ai_laws/ai_law = al
+		if(initial(ai_law.id) == lawid)
+			return ai_law
+	return null
+
+/datum/ai_laws/united_nations/add_inherent_law(law)
+	return //nuh uh
+
+/datum/ai_laws/united_nations/add_ion_law(law)
+	return //nope!
+
+/datum/ai_laws/united_nations/add_hacked_law(law)
+	return //nice try (emagging borgs still hard replaces this lawset though, and that's fine.)
+
+// this is in the server config lawsets otherwise i would remove it
+/datum/ai_laws/toupee
```

Comment:
Head admin not maintainer opinion: Honestly just straight up delete this until another head admin tells me not to let you do that

Reply:
👍